akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -318,55 +318,32 @@ private[akka] final case class Grouped[T](n: Int) extends PushPullStage[T, immut
  * INTERNAL API
  */
 private[akka] final case class Sliding[T](n: Int, step: Int) extends PushPullStage[T, immutable.Seq[T]] {
-  private val buf = {
-    val b = Vector.newBuilder[T]
-    b.sizeHint(n)
-    b
-  }
-  var bufferedElements = 0
+  private var buf = Vector.empty[T]
 
   override def onPush(elem: T, ctx: Context[immutable.Seq[T]]): SyncDirective = {
-    buf += elem
-    bufferedElements += 1
-    if (bufferedElements < n) {
+    buf :+= elem
+    if (buf.size < n) {
+      ctx.pull()
+    } else if (buf.size == n) {
+      ctx.push(buf)
+    } else if (step > n) {
+      if (buf.size == step)
+        buf = Vector.empty
```

Comment:
Also, speaking of building up a huge vector for no reason, look at the occurrences of `buf.result()` in the code I replaced. *shrug*